pkg/ebpf/c/tracee.bpf.c

```diff
@@ -499,56 +499,76 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
     if (!init_program_data(&p, ctx))
         return 0;
 
-    // Note: we don't place should_trace() here, so we can keep track of the cgroups in the system
-    struct task_struct *parent = (struct task_struct *) ctx->args[0];
+    // no should_trace() here so we can keep track of the cgroups in the system
+
     struct task_struct *child = (struct task_struct *) ctx->args[1];
+    struct task_struct *leader = get_leader_task(child);
+    struct task_struct *parent = get_leader_task(get_parent_task(leader));
+
+    // - tgid in kernel is pid in user space
+    // - pid in kernel is tid in user space
 
-    u64 start_time = get_task_start_time(child);
+    // All information here is also important to the process tree (analyze mode)
+
+    // Parent here is always the real process (not a thread)
+    u64 parent_start_time = get_task_start_time(parent);
+    int parent_pid = get_task_host_tgid(parent);
+    int parent_tid = get_task_host_pid(parent);
+    int parent_ns_pid = get_task_ns_tgid(parent);
+    int parent_ns_tid = get_task_ns_pid(parent);
+    // If the "real parent" is a thread, we need to get the leader's info
+    u64 leader_start_time = get_task_start_time(leader);
+    int leader_pid = get_task_host_tgid(leader);
+    int leader_tid = get_task_host_pid(leader);
+    int leader_ns_pid = get_task_ns_tgid(leader);
+    int leader_ns_tid = get_task_ns_pid(leader);
+    // If the "real parent" parent is a thread, we need to get the child's info as well
```

Comment:
Same thing you did in the previous comment: the task (process or thread) that did the fork. 
Parent = the process that generated the thread group which "current" is part of
Leader = the leader of the thread group which "current" is part of (it could even be itself)
Child = current task (if a process, or the initial thread, leader == child, if a thread, leader != child)
This meaning is already base for the process tree and something Alon is currently working with.